api/v1beta1/azuremanagedcontrolplane_webhook.go

```diff
@@ -96,6 +96,12 @@ func (mw *azureManagedControlPlaneWebhook) Default(ctx context.Context, obj runt
 		ctrl.Log.WithName("AzureManagedControlPlaneWebHookLogger").Error(err, "setDefaultSSHPublicKey failed")
 	}
 
+	// PaidManagedControlPlaneTier has been replaced with StandardManagedControlPlaneTier since v2023-02-01.
+	if m.Spec.SKU != nil && m.Spec.SKU.Tier == PaidManagedControlPlaneTier {
+		m.Spec.SKU.Tier = StandardManagedControlPlaneTier
+		ctrl.Log.WithName("AzureManagedControlPlaneWebHookLogger").Info("Paid SKU tier is deprecated and has been replaced by Standard")
```

Comment:
> We can even use a webhook warning to warn the user that "Paid" is deprecated and has been replaced by "Standard" now that those are available

I think what Cecile was referring to here was to include this message in the `admission.Warnings` returned by `ValidateCreate`. It looks like controller-runtime doesn't allow us to return those warnings from the defaulting webhook.

Reply:
That's right. However, if we're changing the value in the defaulting webhook, we'll never print the warning since the validation webhook runs after defaulting (so by the time we reach `ValidateCreate` we would have already changed the value). 

I think the way it's implemented now is probably good enough, wdyt @nojnhuh ?

Reply:
It seems like it wouldn't be too messy to do the validation in the webhook and the Paid->Standard translation further down the road, but I agree keeping everything in the defaulting webhook works just as well.